patroni/psycopg.py

```diff
@@ -14,27 +22,80 @@
     except ImportError:
         _legacy = True
 
-    def quote_literal(value, conn=None):
+    if TYPE_CHECKING:
+        from psycopg2.extensions import connection
+
+    def quote_literal(value: Any, conn: Optional["connection"] = None) -> str:
+        """Quote *value* as a SQL literal.
+
+        .. note::
+            *value* is quoted through ``psycopg`` adapters.
+
+        :param value: value to be quoted.
+        :param conn: if a connection is given then :func:`quote_literal` checks if any special handling based on server
+            parameters needs to be applied to *value* before quoting it as a SQL literal.
+
+        :returns: *value* quoted as a SQL literal.
+        """
         value = adapt(value)
         if conn:
             value.prepare(conn)
         return value.getquoted().decode('utf-8')
 except ImportError:
-    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError
+    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError, Connection
+
+    def _connect(*args: Any, **kwargs: Any) -> Connection[Any]:
+        """Call ``psycopg.connect`` with ``*args`` and ``**kwargs``.
+
+        .. note::
+            Will create ``server_version`` attribute in the returning connection, so it keeps compatibility with the
+            object that would be returned by ``psycopg2.connect``.
+
+        :param args: positional arguments to call ``psycopg.connect`` with.
+        :param kwargs: keyword arguments to call ``psycopg.connect`` with.
 
-    def _connect(*args, **kwargs):
+        :returns: a connection to the database.
+        """
         ret = __connect(*args, **kwargs)
-        ret.server_version = ret.pgconn.server_version  # compatibility with psycopg2
+        setattr(ret, 'server_version', ret.pgconn.server_version)  # compatibility with psycopg2
         return ret
 
-    def _quote_ident(value, conn):
+    def _quote_ident(value: Any, conn: Connection[Any]) -> str:
```

Comment:
```suggestion
    def _quote_ident(value: Any, conn: Connection) -> str:
```

Reply:
I applied the suggestion, but by removing `[Any]` I started getting this again in `pyright`:

```
  /Users/israel.barth/Repositories/edb/edb-patroni/patroni/psycopg.py:60:34 - error: Type of parameter "conn" is partially unknown
    Parameter type is "Connection[Unknown]" (reportUnknownParameterType)
```

Reply:
Strange. I don't have it:
```bash
$ pyright patroni/psycopg.py
No configuration file found.
No pyproject.toml file found.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.302
/home/akukushkin/git/patroni/patroni/psycopg.py
  /home/akukushkin/git/patroni/patroni/psycopg.py:13:26 - error: "__version__" is unknown import symbol (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.764sec

```

Reply:
Strange indeed! I'm running the very same `pyright` version.

Reply:
fwiw I ran pyright on that file with all supported python versions and don't see that "partially unknown" message